core/src/hir/type_context.rs

```diff
@@ -68,6 +74,7 @@ pub enum TypeId {
 pub enum SymbolId {
     TypeId(TypeId),
     TraitId(TraitId),
+    FunctionId(FunctionId),
```

Comment:
nit: please document on `hir::Method` that it can be used for a function or a method